### PR TITLE
Add validate action

### PR DESCRIPTION
**Pull request by thodson-usgs**
Adds an action to validate terraform on pushes and pull requests.

### CODE REVIEW

File: .github/workflows/backend_override.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+terraform {
```
Comment (yuvipanda) [inline .github/workflows/backend_override.tf:1]: Can you create a directory called `ci` and move this into that? I think leaving only github workflows in the workflow directory is cleaner.

File: .github/workflows/validate-terraform.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  validate-terraform:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Terraform repo
+        uses: actions/checkout@v3
+
+      - name: Configure local backend
+        run: |
+          cp .github/workflows/backend_override.tf terraform/aws/
```
Comment (yuvipanda) [inline .github/workflows/validate-terraform.yml:17]: Instead of this, you can pass `-backend-config` to `terraform init`

**Review by yuvipanda — CHANGES_REQUESTED** [review]
Some suggested changes but after that this one is good to go!

File: .github/workflows/validate-terraform.yml (hunk 2 above)
Comment (thodson-usgs) [reply .github/workflows/validate-terraform.yml]: I thought about this, but I'm not sure how to accomplish it here. 
The argument to `-backend-config` is a key-value vars file or "key=value". You can't pass a terraform configuration file, so the `terraform` block and `backend` setting must be part of the current configuration.  For the validation action, we change the backend type from `s3` to `local`.  I wasn't able to do that via `-backend-config` so I used the override.

File: .github/workflows/backend_override.tf (hunk 1 above)
Comment (thodson-usgs) [reply .github/workflows/backend_override.tf]: Done

**Review by thodson-usgs — COMMENTED** [review]
Thanks, @yuvipanda. Maybe you know a solution for backend-config, but I couldn't figure it out in this case.

**Comment by thodson-usgs** [comment]
@yuvipanda,
I don't think `-bankend-config` will work for this case. Please advise.

**Comment by thodson-usgs** [comment]
@ranchodeluxe, would you mind picking up this review?

This PR would add an action to run `terraform validate`. Best practice might be to run the full `apply & destroy`, but there are several reasons to avoid that in this case.

**Comment by ranchodeluxe** [comment]
> @ranchodeluxe, would you mind picking up this review?
> 
> This PR would add an action to run `terraform validate`. Best practice might be to run the full `apply & destroy`, but there are several reasons to avoid that in this case.
> 
> As far as I'm aware, this is good to go and tested on my fork.

Yeah, unless a repo is private I tend to not use `plan || apply` in GH actions and if then in separate steps to avoid bad juju.

**Review by ranchodeluxe — APPROVED** [review]
looks good!

File: ci/backend_override.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+terraform {
```
[the remark this answers is not in the document]
Comment (thodson-usgs) [reply ci/backend_override.tf:1]: Still not sure whether `--backend-config` can swap backend type from `s3` to `local`, but I agree that would be ideal. 

Thanks @ranchodeluxe

**Pull request merged**